### PR TITLE
[Calcium]: New version

### PR DESCRIPTION
**Pull request by thofma**
Build against the newest Arb with compat crudges removed.

### CODE REVIEW

File: C/Calcium/build_tarballs.jl [hunk 1]
```diff
@@ -56,23 +54,23 @@ make install LIBDIR=$(basename ${libdir})
...
 # Dependencies that must be installed before this package can be built
-f(v::VersionNumber) = build_for_julia16_or_newer ?  VersionNumber(v.major, v.minor, v.patch + 1) : v
+
 dependencies = [
-    Dependency("FLINT_jll"; compat = "~$(f(v"200.800.100"))"),
-    Dependency("Arb_jll", compat = "~$(f(v"200.2000"))"),
-    Dependency("Antic_jll", compat = "~$(f(v"0.200.500"))"),
-    Dependency("GMP_jll", build_for_julia16_or_newer ? v"6.2.0" : v"6.1.2"),
-    Dependency("MPFR_jll", build_for_julia16_or_newer ? v"4.1.1" : v"4.0.2"),
+    Dependency("FLINT_jll"; compat = "~$(v"200.800.101")"),
```
Comment (thofma) [inline C/Calcium/build_tarballs.jl:67]: @fingolfin Do we actually still need the `compat` entries? I always forget what is actually going on here.

File: C/Calcium/build_tarballs.jl [hunk 2]
```diff
@@ -56,23 +54,23 @@ make install LIBDIR=$(basename ${libdir})
...
+    Dependency("Arb_jll", compat = "~$(v"200.2100.101")"),
+    Dependency("Antic_jll", compat = "~$(v"0.200.501")",
```
Comment (thofma) [inline C/Calcium/build_tarballs.jl:69]:
```suggestion
    Dependency("Antic_jll", compat = "~$(v"0.200.501")"),
```

File: C/Calcium/build_tarballs.jl [hunk 3]
```diff
@@ -23,11 +23,9 @@ import Pkg.Types: VersionSpec
...
 name = "Calcium"
 upstream_version = v"0.4.1"
-build_for_julia16_or_newer = true
-version_offset = build_for_julia16_or_newer ? v"0.0.1" : v"0.0.0"
-version = VersionNumber(upstream_version.major * 100 + version_offset.major,
-                        upstream_version.minor * 100 + version_offset.minor,
-                        upstream_version.patch * 100 + version_offset.patch)
+version = VersionNumber(upstream_version.major * 100,
+                        upstream_version.minor * 100,
+                        upstream_version.patch * 100 + 1) # remove the 1 once the upstream_version is changed
```
Comment (fingolfin) [inline C/Calcium/build_tarballs.jl:28]: We had the `version_offset` to be able to deal with rebuilds of the package. Of course you can manually add/insert the `+1` (or +2, or ...); I personally find it easier to read with the `version_offset` in place... But as usual, pure taste *shrug*

File: C/Calcium/build_tarballs.jl (hunk 1 above)
Comment (fingolfin) [reply C/Calcium/build_tarballs.jl:67]: Yes we want those to make sure that only matching versions are used together. After all, this JLL won't be compatible with an older or much newer FLINT_JLL (it hopefully will be compatible with FLINT 200.800.200 etc. but probably not with 200.900.x)

**Review by fingolfin — APPROVED** [review]

File: C/Calcium/build_tarballs.jl (hunk 3 above)
Comment (thofma) [reply C/Calcium/build_tarballs.jl:28]: Ah, I thought it was specifically for the < 1.6 dichotomy. I will add it back.

**Comment by thofma** [comment]
Good to go from my part.

File: C/Calcium/build_tarballs.jl (hunk 1 above)
Comment (thofma) [reply C/Calcium/build_tarballs.jl:68]: Thanks for the explanation!

**Pull request merged**